packages/client/hmi-client/src/components/code/tera-code.vue

```diff
@@ -131,13 +192,27 @@ function onSelectedTextChange() {
 	selectedText.value = editor.value?.getSelectedText() ?? '';
 }
 
+function setFileExtension(fileName: string) {
+	// check name for file extension
+	// if there is no extension, add the appropriate one based on the selected language
+	const name = fileName.split('.');
+	if (name.length > 0) {
+		return name[0].concat('.').concat(getFileExtension(programmingLanguage.value));
+	}
+	return fileName;
+}
```

Comment:
I think it would be better off to have this function in utilities or services, would be easier to unit test than being inside a component.